executor/executor_test.go

```diff
@@ -92,16 +106,46 @@ func (d *dummySDAPIBroken) PostCommand(specPath string, smallSpec *util.CommandS
 }
 
 func dummyAPICommand(format string) (cmd *util.CommandSpec) {
-	cmd = &util.CommandSpec{
-		Namespace:   dummyNameSpace,
-		Name:        dummyName,
-		Description: dummyDescription,
-		Version:     dummyVersion,
-		Format:      format,
+	switch format {
```

Comment:
`Namespace` .. `Format` propertie's settings are still redundant. How about the following code.

```go
cmd = &util.CommandSpec{
	Namespace:   dummyNameSpace,
	Name:        dummyName,
	Description: dummyDescription,
	Version:     dummyVersion,
	Format:      format,
}

switch format {
case binaryFormat:
	cmd.Binary = &util.Binary{
		File: dummyFile,
	}
case habitatFormat:
	cmd.Habitat = &util.Habitat{
		Mode:    dummyMode,
		Package: dummyPackage,
		Command: dummyCommand,
	}
# ...
default:
	return nil
}

return cmd
```

Reply:
Looks good!